geocruncher/main.py

```diff
@@ -17,26 +17,59 @@ def main():
     run_geocruncher(sys.argv)
 
 def run_geocruncher(args):
-
     model = GeologicalModel(args[3])
     model.topography = txt_extract(args[4])
     box = model.getbox()
     
     if args[1] == 'all':
         with open(args[2]) as f:
             data = json.load(f)
-        nPoints=30
+        nPoints=40
         crossSections = []
         for rect in data:
             xCoord=[rect["lowerLeft"]["x"], rect["upperRight"]["x"]]
             yCoord=[rect["lowerLeft"]["y"], rect["upperRight"]["y"]]
             zCoord=[rect["lowerLeft"]["z"], rect["upperRight"]["z"]]
+            if zCoord[0] < box.zmax or zCoord[0] > box.zmin or zCoord[1] > box.zmin or zCoord[1] < box.zmax: 
+                if isOutofBounds(xCoord[0], yCoord[0], box) == 'true':
+                    (xCoord[0], yCoord[0]) = intersectBounds(xCoord, yCoord, zCoord, box, 0)
+                if isOutofBounds(xCoord[1], yCoord[1], box) == 'true':
+                    (xCoord[1], yCoord[1]) = intersectBounds(xCoord, yCoord, zCoord, box, 1)
+            if xCoord[0] == xCoord[1]:
+                xCoord[0] = xCoord[1] + 1
             crossSections.append(CrossSectionIntersections.output(xCoord,yCoord,zCoord,nPoints,model, [1, 1]));
         xCoord=[box.xmin,box.xmax]
         yCoord=[box.ymin,box.ymax]
-        nPoints=45
+        nPoints=60
         outputs = { 'forMaps': MapIntersections.output(xCoord,yCoord,nPoints,model), 'forCrossSections': crossSections }
         with open(args[5], 'w') as f:
             json.dump(outputs, f, indent = 2, cls=GeocruncherJsonEncoder)
         sys.stdout.flush()
-    
+
+def isOutofBounds(xCoord, yCoord, box):
+    if xCoord > box.xmax or xCoord < box.xmin or yCoord > box.ymax or yCoord < box.ymin:
+        return 'true'
+    else :
+        return 'false'
+
+def intersectBounds(xCoord, yCoord, zCoord, box, index):
+    slope = (yCoord[1] - yCoord[0]) /  (xCoord[1] - xCoord[0])

```

Comment:
You are saying this does not change the coordinates, so (0,0) is still the left-lower corner of the full CS image and (1,1) is the top right corner of the full CS image? Then what does this do? Please document the method.

Reply:
Ok, I did not understood your question then.

My answer was that it did not change coordinates especially because it was still between 0-1. 
But I agree that it is between 0-1 on the reduce CS and not on the whole CS. 

I will do the coordinates change here and let the frontEnd unchanged, I guess this is what you prefer.

Reply:
I assume it is less work to not change the coordinate system itself. However, I am fine with filtering out-of-project intersections here. As you correctly pointed out, this could be better for performance.

Reply:
I made it so now so we only calculate in the project boundaries but still return relative to the whole CS